common/bolt11.c

```diff
@@ -678,6 +679,28 @@ static bool bech32_decode_alloc(const tal_t *ctx,
 	return true;
 }
 
+const char *to_canonical_invstr(const tal_t *ctx,
+				const char *invstring)
+{
+	/* BOLT #11:
+	 *
+	 * If a URI scheme is desired, the current recommendation is to either
+	 * use 'lightning:' as a prefix before the BOLT-11 encoding */
+	if (strstarts(invstring, "lightning:") || strstarts(invstring, "LIGHTNING:"))
+		invstring += strlen("lightning:");;
+	return tal_strlowering(ctx, invstring);
+}
+
+inline static bool is_canonical_invstr(const char *invstring)
+{
+	/* BOLT #11:
+	 *
+	 * If a URI scheme is desired, the current recommendation is to either
+	 * use 'lightning:' as a prefix before the BOLT-11 encoding */
+	return !(strstarts(invstring, "lightning:") ||
+		strstarts(invstring, "LIGHTNING:"));
+}
```

Comment:
This should be inverted and called "has_lightning_prefix()"?

Reply:
Much better yeah this logic was a little bit brain stressing :)